tests/MongoDB.Driver.Examples/ChangeStreamExamples.cs

```diff
@@ -107,19 +108,18 @@ public void ChangeStreamExample3()
                 })
                 .Start();
 
-                var enumerator = inventory.Watch().ToEnumerable().GetEnumerator();
-                enumerator.MoveNext();
-                lastChangeStreamDocument = enumerator.Current;
+                previousCursor = inventory.Watch(new ChangeStreamOptions { BatchSize = 1 });
+                while (previousCursor.MoveNext() && previousCursor.Current.Count() == 0) { } // keep calling MoveNext until we've read the first batch
             }
 
             {
                 // Start Changestream Example 3
-                var resumeToken = lastChangeStreamDocument.ResumeToken;
+                var resumeToken = previousCursor.GetResumeToken();
                 var options = new ChangeStreamOptions { ResumeAfter = resumeToken };
-                var enumerator = inventory.Watch(options).ToEnumerable().GetEnumerator();
-                enumerator.MoveNext();
-                var next = enumerator.Current;
-                enumerator.Dispose();
+                var cursor = inventory.Watch(options);
+                cursor.MoveNext();
```

Comment:
This only works because in the setup to the test we ensured that a change document would be pending.

In general, MoveNext might return false and the call to First in the next statement could throw an exception.

Again... this is because these examples are fundamentally flawed.